inc/import.class.php

```diff
@@ -0,0 +1,1220 @@
+    * Have I the global right to "view" the Object
+    *
+    * Default is true and check entity if the objet is entity assign
+    *
+    * May be overloaded if needed
+    *
+    * @return booleen
+    **/
+   static function canView() {
+      return Session::haveRight(self::$rightname, READ);
+   }
+
+   /**
+    * Have I the global right to "create" the Object
+    * May be overloaded if needed (ex KnowbaseItem)
+    *
+    * @return booleen
+    **/
+   static function canUpdate() {
+      return Session::haveRightsOr(self::$rightname, [UPDATE]);
+   }
+
+   /**
+    * Clean object veryfing criteria (when a relation is deleted)
+    *
+    * @param $crit array of criteria (should be an index)
+    */
+   public function clean($crit) {
+      global $DB;
+
+      foreach ($DB->request($this->getTable(), $crit) as $data) {
+         $this->delete($data);
+      }
+   }
+
+   /**
+    * Get Tab Name used for itemtype
+    *
+    * NB : Only called for existing object
+    *      Must check right on what will be displayed + template
+    *
+    * @since 0.83
+    *
+    * @param CommonGLPI $item Item on which the tab need to be displayed
+    * @param boolean    $withtemplate is a template object ? (default 0)
+    *
+    * @return string tab name
+    **/
+   function getTabNameForItem(CommonGLPI $item, $withtemplate = 0) {
+
+      if (!$withtemplate) {
+         if ($item->getID() && $this->canView()) {
+            return self::getTypeName(2);
+         }
+      }
+      return '';
+   }
+
+   public function rawSearchOptions() {
+      $tab = [];
+
+      $tab[] = [
+         'id'   => 'common',
+         'name' => __('Characteristics')
+      ];
+
+
+      //Données de l'import External
+      $tab[] = [
+         'id'            => '1',
+         'table'         => $this->getTable(),
+         'field'         => 'id',
+         'name'          => __('ID'),
+         'massiveaction' => false,
+         'datatype'      => 'number'
+      ];
+
+      $tab[] = [
+         'id'            => '2',
+         'table'         => $this->getTable(),
+         'field'         => 'id_external',
+         'name'          => __('ID External', 'resources'),
+         'massiveaction' => false,
+         'datatype'      => 'text'
+      ];
+
+      $tab[] = [
+         'id'            => '3',
+         'table'         => $this->getTable(),
+         'field'         => 'name',
+         'name'          => __('Name'),
+         'massiveaction' => false,
+         'datatype'      => 'text'
+      ];
+
+      $tab[] = [
+         'id'            => '4',
+         'table'         => $this->getTable(),
+         'field'         => 'firstname',
+         'name'          => __('Firstname'),
+         'massiveaction' => false,
+         'datatype'      => 'text'
+      ];
+
+      $tab[] = [
+         'id'            => '5',
+         'table'         => $this->getTable(),
+         'field'         => 'matricule',
+         'name'          => __('Matricule'),
```

Comment:
Check in file fr_FR.po, the translation of "Matricule" is "Administrative number"